CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSVCore.java

```diff
@@ -55,24 +55,27 @@ public void handleMessage(ExtensionServiceMessage message) {
             }
             
             oConfigHandler.configComplete = false;
-                        
-            CompletableFuture<Boolean> success = client.connectToSource();
-            
-            try {
-                if ( !success.get(10, TimeUnit.SECONDS) ) {
-                    if (!client.isOpen()) {
-                        log.error("Failed to connect to server url '{}'.", targetVantiqServer);
-                    } else if (!client.isAuthed()) {
-                        log.error("Failed to authenticate within 10 seconds using the given authentication data.");
-                    } else {
-                        log.error("Failed to connect within 10 seconds");
+
+            // Spin off a thread to handle doing the reconnect and error/log handling, this way the handler doesn't
+            // block the onMessage() handler in the ExtensionWebSocketListener from processing future messages
+            new Thread( () -> {
+                CompletableFuture<Boolean> success = client.connectToSource();
```

Comment:
Could we put these forking the thread & work into a utility procedure or something in the SDK?  Then, each connector can handle reconnects by via

```
    // Do my connector special stuff
   client.handleReconnect() 

   // It may have to expose the configComplete or define an "andThen" type thing to set the configComplete when done.
```

Just seems like a lot of duplicated code...